src/common/options/crimson.yaml.in

```diff
@@ -22,7 +22,7 @@ options:
 - name: crimson_seastar_smp
   type: uint
   level: advanced
-  desc: Number of seastar reactor threads to use for the osd
+  desc: Number of seastar reactor threads to use for each osd
```

Comment:
Probably not a reason to block merging this PR to clean up "N_CORES_FOR_SEASTAR", but ...

Looks we are setting "seastar --smp" through this option `crimson_seastar_smp` and also through `main_config_bootstrap_helpers.h`, but according to https://github.com/ceph/ceph/blob/main/src/vstart.sh#L1118, we are setting "seastar --cpuset" in an inconsistent way without the bootstrap helper.

My question is: should we replace this option by `crimson_seastar_cpu_cores` which follows the same format of `crimson_alien_thread_cpu_cores` below, and instead, deduce the "seastar --smp" from  `crimson_seastar_cpu_cores`?

Reply:
This could work, although the only complexity here would be that we'll need to "divide" `crimson_seastar_cpu_cores` by the OSD count in order to deduce seastar::smp. Did I understand you correctly?

Reply:
> we'll need to "divide" crimson_seastar_cpu_cores by the OSD count in order to deduce seastar::smp

I'm not quite following your point. For example, if `crimson_seastar_cpu_cores` is set to range `4-6`, `seastar::smp` can be deduced as `6 - 4 + 1` which is 3, without the need to "divide" by the OSD count, right ?

Reply:
Are you suggesting to set `crimson_seastar_cpu_cores` for each OSD?
My intent was that if we set the range to be `4-6` in a 3 OSD cluster, each OSD will have a smp of 1. If only a single OSD is deployed it will get all the range.

Reply:
The current `crimson_seastar_smp` results in `crimson_seastar_smp` * `OSD count` seastar threads.
The issue with setting single `crimson_seastar_cpu_cores` (e.g 4-6) would mean that either we need to set it for each OSD or we need to divide the given cores by each one.

Reply:
@Matan-B I'm not quite following, are you referring to vstart.sh?  OSD configs generally pertain only to that OSD, it would be the deployment mechanism's job to figure out how to apportion core between OSDs and set the individual OSD configs appropriately.

Reply:
> Are you suggesting to set `crimson_seastar_cpu_cores` for each OSD?

Yes, I suppose it is a per-OSD configuration. If there are 3 OSDs in the same host using 9 cores, `crimson_seastar_cpu_cores` would be "0-2", "3-5", "6-8" for each OSD.

> The current `crimson_seastar_smp` results in `crimson_seastar_smp` * OSD count seastar threads.

So `crimson_seastar_smp` is also a per-OSD configuration -- why `crimson_seastar_cpu_cores` would be different as a configuration ?

> OSD configs generally pertain only to that OSD, it would be the deployment mechanism's job to figure out how to apportion core between OSDs and set the individual OSD configs appropriately.

I agree.

Reply:
I'm not familiar with deployments. Is it reasonable to expose a ceph configuration such as `crimson_seastar_cpu_cores` that will end up with different values for every OSD?

If go with `crimson_seastar_smp`, it seems simpler to set it to a global "3" in the above example. But we still need to deal with cpuset isolation somehow.